sdk/appconfiguration/azure-appconfiguration/azure/appconfiguration/_models.py

```diff
@@ -39,29 +42,362 @@ class ConfigurationSetting(Model):
+    :type content_type: str
+    :ivar last_modified:
+    :vartype last_modified: datetime
+    :ivar read_only:
+    :vartype read_only: bool
+    :param tags:
+    :type tags: dict[str, str]
+    """
+
+    _attribute_map = {
+        "etag": {"key": "etag", "type": "str"},
+        "key": {"key": "key", "type": "str"},
+        "label": {"key": "label", "type": "str"},
+        "content_type": {"key": "content_type", "type": "str"},
+        "value": {"key": "value", "type": "str"},
+        "last_modified": {"key": "last_modified", "type": "iso-8601"},
+        "read_only": {"key": "read_only", "type": "bool"},
+        "tags": {"key": "tags", "type": "{str}"},
+    }
+    key_prefix = ".appconfig.featureflag/"
+    _feature_flag_content_type = (
+        "application/vnd.microsoft.appconfig.ff+json;charset=utf-8"
+    )
+    kind = "FeatureFlag"
+
+    def __init__(self, key, enabled, filters=None, **kwargs):
+        # type: (str, bool, Optional[List[Dict[str, Any]]]) -> None
+        self._enabled = enabled
+        super(FeatureFlagConfigurationSetting, self).__init__(**kwargs)
+        self.key = key
+        if not key.startswith(self.key_prefix):
+            self.key = self.key_prefix + key
+        self.label = kwargs.get("label", None)
+        self.content_type = kwargs.get("content_type", self._feature_flag_content_type)
+        self.last_modified = kwargs.get("last_modified", None)
+        self.tags = kwargs.get("tags", {})
+        self.read_only = kwargs.get("read_only", None)
+        self.etag = kwargs.get("etag", None)
+        self.description = kwargs.get("description", None)
+        self.display_name = kwargs.get("display_name", None)
+        self._filters = filters or []
+        self._value = kwargs.get("value", {})
+
+    @property
+    def enabled(self):
+        # type: () -> bool
+        if self._value is None:
+            self._value = {"enabled": self._enabled}
+        elif not isinstance(self._value, dict):
+            raise AttributeError("value parameter is expected to be a dictionary")
+        else:
+            self._value["enabled"] = self._enabled
+        return self._enabled
+
+    @enabled.setter
+    def enabled(self, value):
+        # type: (bool) -> bool
+        if self._value is None:
+            self._value = {"enabled": value}
+        elif not isinstance(self._value, dict):
+            raise AttributeError("value parameter is expected to be a dictionary")
+        else:
+            self._value["enabled"] = self._enabled
+        self._enabled = value
+
+    @property
+    def value(self):
+        # type: () -> Dict[str, Any]
+        self._enabled = self._value.get("enabled") or self._enabled
+        return self._value
+
+    @value.setter
+    def value(self, new_value):
+        # type: (Dict[str, Any]) -> None
+        if not new_value:
+            new_value = {}
+        self._enabled = new_value.get("enabled") or self._enabled
+        self._value = new_value
```

Comment:
Looks like if we set value=None then get enabled, the value will be changed to {"enabled": self._enabled}?

Reply:
if `new_value = None` this will raise an `AttributeError` on line 196 (`new_value.get("enabled")`).

Reply:
{}.get('enabled')

returns None. :)

Reply:
Ahh, good catch. I will remove the `if not new_value` lines, then it will throw an `AttributeError`.